test/unit/migrate_config_proxy_to_http_proxy.rb

```diff
@@ -0,0 +1,167 @@
+require 'test_plugin_helper'
+require File.join(Pathname.new(File.dirname(__FILE__)).parent.parent, 'db', 'migrate', '20200728054557_refer_config_proxy_to_http_proxy.rb')
+
+module ForemanVirtWhoConfigure
+  class MigrateConfigProxyToHttpProxy < ActiveSupport::TestCase
+    let(:migration) { klass.new }
+
+    def migration_has_been_run?(version)
+      ActiveRecord::SchemaMigration.where(version: version).exists?
+    end
+
+    describe "up" do
+      let(:config) { FactoryBot.build(:virt_who_config) }
+
+      before do
+        if migration_has_been_run?('20200728054557')
+          migration.down
+        end
+
+        HttpProxy.delete_all
+      end
+
+      it "should create Http Proxy if already not exist" do
+        config.proxy = "test.example.com"
+        config.save!
+
+        assert_difference('HttpProxy.count', 1) do
+          migration.up
+        end
+
+        assert_equal "http://test.example.com", created_http_proxy.full_url
+      end
+
+      it "should create Http Proxy with user and password if already not exist" do
+        config.proxy = "http://user:password@test.example.com"
+        config.save!
+
+        assert_difference('HttpProxy.count', 1) do
+          migration.up
+        end
+
+        assert_equal "http://user:password@test.example.com", created_http_proxy.full_url
+        assert_equal "user", created_http_proxy.username
+        assert_equal "password", created_http_proxy.password
+      end
+
+      it "should create valid Http Proxy for invalid proxy if already not exist" do
+        config.proxy = "test.example.com"
+        config.save!
+
+        assert_difference('HttpProxy.count', 1) do
+          migration.up
+        end
+
+        assert_equal "http://test.example.com", created_http_proxy.full_url
+      end
+
+      it "should not create Http Proxy for invalid config proxy if already not exist" do
+        config.proxy = "test.example.com:5000"
```

Comment:
I think we should end up with an HttpProxy here with full_url: http://test.example.com:5000
What do you think?